File: app/src/main/java/com/ketanolab/simidic/DiccionariesActivity.java
package com.ketanolab.simidic;

import java.util.ArrayList;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;

import android.view.Menu;
import android.view.MenuItem;


import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.ketanolab.simidic.adapters.DictionariesViewPagerAdapter;
import com.ketanolab.simidic.util.Dictionaries;
import com.ketanolab.simidic.viewpager.CirclePageIndicator;

public class DiccionariesActivity extends AppCompatActivity {

	// Paginado
	private ViewPager viewPager;
	private CirclePageIndicator indicator;
	private DictionariesViewPagerAdapter adaptadorViewPager;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_dictionaries);
		getSupportActionBar().setDisplayHomeAsUpEnabled(true);

		// Paginado
		viewPager = (ViewPager) findViewById(R.id.viewPager);
		adaptadorViewPager = new DictionariesViewPagerAdapter(this);
		viewPager.setAdapter(adaptadorViewPager);
		indicator = (CirclePageIndicator) findViewById(R.id.indicator);
		indicator.setViewPager(viewPager);
		// Fin Paginado
		getDictionaries();


	}

	public void getDictionaries() {
		ArrayList<String> paths = Dictionaries.scanDictionariesAndValidation(this);
		for (int i = 0; i < paths.size(); ++i) {
			SQLiteDatabase db = SQLiteDatabase.openOrCreateDatabase(paths.get(i), null);
			setDictionaryData(db);
			db.close();
		}
	}
	private void setDictionaryData(SQLiteDatabase db) {
		Cursor cursor = db.rawQuery("SELECT author, language_begin, language_end, description FROM info", null);
		if (cursor.moveToFirst()) {
			String author = cursor.getString(0);
			String beginLanguage = cursor.getString(1);
			String endLanguage = cursor.getString(2);
			String description = cursor.getString(3);
			cursor.close();
			adaptadorViewPager.adicionarItem(R.drawable.img_dictionary, beginLanguage + " - " + endLanguage, author,
					description);
			adaptadorViewPager.notifyDataSetChanged();
		}
		cursor.close();
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		if (item.getItemId() == android.R.id.home) {
			this.finish();
		}
		return true;
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {

		return true;

	}

}

File: app/src/main/java/com/ketanolab/nusimi/DiccionariesActivity.kt
package com.ketanolab.nusimi

import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.ketanolab.nusimi.adapters.DictionariesViewPagerAdapter
import com.ketanolab.nusimi.util.Dictionaries
import com.ketanolab.nusimi.viewpager.CirclePageIndicator

class DiccionariesActivity : BaseActivity() {
    // Paginado
    private var viewPager: ViewPager? = null
    private var indicator: CirclePageIndicator? = null
    private var adaptadorViewPager: DictionariesViewPagerAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupLanguage()
        setContentView(R.layout.activity_dictionaries)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.title = getString(R.string.dictionaries)
        // Paginado
        viewPager = findViewById<View>(R.id.viewPager) as ViewPager
        adaptadorViewPager = DictionariesViewPagerAdapter(this)
        viewPager!!.adapter = adaptadorViewPager
        indicator = findViewById<View>(R.id.indicator) as CirclePageIndicator
        indicator!!.setViewPager(viewPager)
        // Fin Paginado
        dictionaries
    }

    val dictionaries: Unit
        get() {
            val paths = Dictionaries.scanDictionariesAndValidation(this)
            for (i in paths.indices) {
                val db = SQLiteDatabase.openOrCreateDatabase(paths[i], null)
                setDictionaryData(db)
                db.close()
            }
        }

    private fun setDictionaryData(db: SQLiteDatabase) {
        val cursor =
            db.rawQuery("SELECT author, language_begin, language_end, description FROM info", null)
        if (cursor.moveToFirst()) {
            val author = cursor.getString(0)
            val beginLanguage = cursor.getString(1)
            val endLanguage = cursor.getString(2)
            val description = cursor.getString(3)
            cursor.close()
            adaptadorViewPager!!.adicionarItem(
                R.drawable.img_dictionary, "$beginLanguage - $endLanguage", author,
                description
            )
            adaptadorViewPager!!.notifyDataSetChanged()
        }
        cursor.close()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return true
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        return true
    }
}